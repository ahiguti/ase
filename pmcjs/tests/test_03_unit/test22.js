
// var x = Perl.NewHash({ foo: 'fooval', bar: 'barval' });
var x = Perl.Eval("{ foo => 'foooo' }");
print("foo: [" + x.foo + "]");
print("x.Detach()");
try {
  x.Detach();
} catch (e) {
  print("err: " + e.error);
}
print("foo: [" + x.foo + "]");
print("Perl.Detach(x)");
Perl.Detach(x);
print("foo: [" + x.foo + "]");
