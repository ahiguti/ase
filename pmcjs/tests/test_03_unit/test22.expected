foo: [foooo]
x.Detach()
err: TypeMismatch
foo: [foooo]
Perl.Detach(x)
foo: [undefined]
